tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -356,6 +356,39 @@ bool DeclResultIdMapper::createStageVars(const DeclaratorDecl *decl,
     if (varId == 0)
       return false;
 
+    // Decorate with interpolation modes for pixel shader input variables
+    if (shaderModel.IsPS() && sigPoint->IsInput()) {
+      const QualType elemType = typeTranslator.getElementType(type);
+
+      if (elemType->isBooleanType() || elemType->isIntegerType()) {
+        // TODO: Probably we can call hlsl::ValidateSignatureElement() for the
+        // following check.
+        if (decl->getAttr<HLSLLinearAttr>() ||
+            decl->getAttr<HLSLCentroidAttr>() ||
+            decl->getAttr<HLSLNoPerspectiveAttr>() ||
+            decl->getAttr<HLSLSampleAttr>()) {
+          emitError("only nointerpolation mode allowed for integer input "
```

Comment:
Basically copied the logic from https://github.com/Microsoft/DirectXShaderCompiler/blob/55667043c389dcba990c678390abc9e7cc9fe1d3/lib/HLSL/DxilValidation.cpp#L3197. They checked both booleans and integers but report error message for only integers. I was trying to keep consistency. 

Besides, booleans are most of the time treated as integers. And I assume if `elemType->isBooleanType()` is true, `elemType->isIntergerType()` will also return true. The above check may be redundant, but it's clear.